src/core/display/Container.js

```diff
@@ -606,17 +606,23 @@ Container.prototype.renderCanvas = function (renderer)
 
 /**
  * Destroys the container
- * @param [destroyChildren=false] {boolean} if set to true, all the children will have their destroy method called as well
+ * @param [opts] {object} Options parameter
+ * @param [opts.children=false] {boolean} if set to true, all the children will have their destroy
+ *      method called as well. 'opts' will be passed on to those calls.
+ *
  */
-Container.prototype.destroy = function (destroyChildren)
+Container.prototype.destroy = function (opts)
 {
+    opts = opts || {};
```

Comment:
I agree with @bigtimebuddy. Only thing I want to clarify is that you shouldn't translate the boolean into an object explicitly. Just pass along the boolean and each object checks if it is true or has its flag to do deletion.

Something along the lines of:

```js
Sprite.prototype.destroy = function (options)
{
    Container.prototype.destroy.call(this, options);
    var destroyTexture = typeof options === 'boolean' ? options : options && options.texture);
    // ....
}
```

That way you destroy the texture if it is `true` or if `options.texture` is true. Same logic for other flags:

```js
var destroyChildren = typeof options === 'boolean' ? options : options && options.children;
```

Maybe someone else can simplify that.